src/lib/chawathe98_matcher.rs

```diff
@@ -265,31 +324,29 @@ impl<T: Clone + Debug + Eq + 'static> MappingStoreGraph<T> {
     /// This overall checks the subtrees of two trees and whether they are the same.
     /// The check is done by checking if the nodes are equal and as well as their descendants.
     pub fn check_subtree(&self, from_node: NodeId<FromNodeId>, to_node: NodeId<ToNodeId>) -> bool {
-        //        let mut output = true;
-        let descendants_from_node = from_node.descendants(&self.from_arena.borrow())
-                                             .collect::<Vec<NodeId<FromNodeId>>>();
-        let descendants_to_node = to_node.descendants(&self.to_arena.borrow())
-                                         .collect::<Vec<NodeId<ToNodeId>>>();
+        let descendants_from_node = from_node.descendants(&self.from_arena.borrow()).collect::<Vec<NodeId<FromNodeId>>>();
+        let descendants_to_node = to_node.descendants(&self.to_arena.borrow()).collect::<Vec<NodeId<ToNodeId>>>();
         if eq(&from_node,
               &self.from_arena.borrow(),
               &to_node,
               &self.to_arena.borrow())
-        {
-            if descendants_from_node.len() == descendants_to_node.len() {
-                for i in 0..descendants_to_node.len() {
-                    if !eq(&descendants_from_node[i],
-                           &self.from_arena.borrow(),
-                           &descendants_to_node[i],
-                           &self.to_arena.borrow())
-                    {
-                        // This means the descendants are !perfect by label and expression
-                        return false;
+            {
+                if descendants_from_node.len() == descendants_to_node.len() {
```

Comment:
From an indenting perspective this would be clearer as:

```rust
if descendants_from_node.len() != descendants_to_node.len() {
    return false;
}
for i in 0..descendants_to_node.len() {
    ...
}
```